internal/tfdiags/consolidate_warnings.go

```diff
@@ -7,112 +7,123 @@ package tfdiags
 			continue
 		}
 
-		warningStats[summary]++
-		if warningStats[summary] == threshold {
+		diagnosticStats[summary]++
+		if diagnosticStats[summary] == threshold {
 			// Initially creating the group doesn't really change anything
-			// visibly in the result, since a group with only one warning
+			// visibly in the result, since a group with only one diagnostic
 			// is just a passthrough anyway, but once we do this any additional
-			// warnings with the same summary will get appended to this group.
-			g := &warningGroup{}
+			// diagnostics with the same summary will get appended to this group.
+			g := &consolidatedGroup{}
 			newDiags = newDiags.Append(g)
-			warningGroups[summary] = g
+			diagnosticGroups[summary] = g
 			g.Append(diag)
 			continue
 		}
 
-		// If this warning is not consolidating yet then we'll just append
+		// If this diagnostic is not consolidating yet then we'll just append
 		// it directly.
 		newDiags = newDiags.Append(diag)
 	}
 
 	return newDiags
 }
 
-// A warningGroup is one or more warning diagnostics grouped together for
+// A consolidatedGroup is one or more diagnostics grouped together for
 // UI consolidation purposes.
 //
-// A warningGroup with only one diagnostic in it is just a passthrough for
+// A consolidatedGroup with only one diagnostic in it is just a passthrough for
 // that one diagnostic. If it has more than one then it will behave mostly
 // like the first one but its detail message will include an additional
-// sentence mentioning the consolidation. A warningGroup with no diagnostics
+// sentence mentioning the consolidation. A consolidatedGroup with no diagnostics
 // at all is invalid and will panic when used.
-type warningGroup struct {
-	Warnings Diagnostics
+type consolidatedGroup struct {
+	Consolidated Diagnostics
 }
 
-var _ Diagnostic = (*warningGroup)(nil)
+var _ Diagnostic = (*consolidatedGroup)(nil)
 
-func (wg *warningGroup) Severity() Severity {
-	return wg.Warnings[0].Severity()
+func (wg *consolidatedGroup) Severity() Severity {
+	return wg.Consolidated[0].Severity()
 }
 
-func (wg *warningGroup) Description() Description {
-	desc := wg.Warnings[0].Description()
-	if len(wg.Warnings) < 2 {
+func (wg *consolidatedGroup) Description() Description {
+	desc := wg.Consolidated[0].Description()
+	if len(wg.Consolidated) < 2 {
 		return desc
 	}
-	extraCount := len(wg.Warnings) - 1
+	extraCount := len(wg.Consolidated) - 1
 	var msg string
+	var diagType string
+	switch wg.Severity() {
+	case Error:
+		diagType = "error"
+	case Warning:
+		diagType = "warning"
+	default:
+		panic(fmt.Sprintf("Invalid diagnostic severity: %#v", wg.Severity()))
+	}
+
 	switch extraCount {
 	case 1:
-		msg = "(and one more similar warning elsewhere)"
+		msg = fmt.Sprintf("(and one more similar %s elsewhere)", diagType)
 	default:
-		msg = fmt.Sprintf("(and %d more similar warnings elsewhere)", extraCount)
+		msg = fmt.Sprintf("(and %d more similar %ss elsewhere)", extraCount, diagType)
```

Comment:
some typo here? 

`%ss` => `%s`

Reply:
I don't think so:

error -> errors
warning -> warnings

Reply:
Ah, I got you.